Refresher/Patching/Patcher.cs

```diff
@@ -123,74 +139,151 @@ private static List<PatchTargetInfo> FindUrl(Stream file)
             byte[] match = new byte[len];
             if (file.Read(match) < len) continue;
             string str = Encoding.UTF8.GetString(match).TrimEnd('\0');
-            
-            if(str.Contains('%')) continue; // Ignore printf strings, e.g. %s
 
-            Regex regex = UrlMatch();
-            MatchCollection matches = regex.Matches(str);
+            if (str.Contains('%')) continue; // Ignore printf strings, e.g. %s
 
-            if (matches.Count != 0)
-                found.Add(new PatchTargetInfo
+            if (UrlMatch().Matches(str).Count != 0)
+                foundUrls.Add(new PatchTargetInfo
                 {
                     Length = len,
                     Offset = foundPosition,
                     Data = str,
                 });
         }
 
-        return found;
+        List<PatchTargetInfo> foundDigests = new();
+
+        foreach (long foundPosition in foundPossibleCookiePositions)
+        {
+            file.Position = foundPosition;
+
+            byte[] cookieBuf = new byte[8];
+            //If we didnt read enough or what we read isnt "cookie\0\0"
+            if (reader.Read(cookieBuf) < cookieBuf.Length || !"cookie\0\0"u8.SequenceEqual(cookieBuf))
+            {
+                //Skip this possibility
+                continue;
+            }
+
+            const int checkSize = 1000;
+
+            //Go back half the check size in bytes (so that `cookie` is in the middle)
+            file.Position -= checkSize / 2;
+
+            byte[] checkArr = new byte[checkSize];
+            Span<byte> toCheck = checkArr.AsSpan().Slice(0, reader.Read(checkArr));
+
+            Regex regex = DigestMatch();
+            for (int i = 0; i < toCheck.Length; i++)
+            {
+                //Skip all null bytes
+                while (i < toCheck.Length && toCheck[i] == 0) i++;
+
+                int len = 0;
+                int start = i;
+
+                //Go over all non-null bytes
+                while (i < toCheck.Length && toCheck[i] != 0)
+                {
+                    len++;
+                    i++;
+                }
```

Comment:
Can't we determine the length with just one loop? Set len to the total length and then subtract

Reply:
im not sure what you mean, the first loop skips all of the null bytes, then the second one only determines length

Reply:
My consciousness level has severely degraded since I've written that comment, and I have forgotten the explanation behind my thought process. Marking as resolved.